sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/common/credentials/SharedKeyCredential.java

```diff
@@ -232,8 +234,8 @@ private String computeHMACSHA256(String stringToSign) {
             String signature = Base64.getEncoder().encodeToString(hmacSha256.doFinal(utf8Bytes));
             return String.format(AUTHORIZATION_HEADER_FORMAT, accountName, signature);
         } catch (NoSuchAlgorithmException | InvalidKeyException ex) {
-            LOGGER.error(ex.getMessage());
-            throw new Error(ex);
+            logger.logAndThrow(new RuntimeException(ex.getMessage()));
```

Comment:
Pass in the exception itself into the constructor and change this to use IllegalStateException as that better aligns with the spec.